### PR TITLE
refs #44: show ParamValues also when do() syntax is used

### PR DESCRIPTION
**Pull request by dk**
Hi,

Here I'm trying to show ParamValues in do() calls, that wasn't possible before as do() and fetch(ParamValues) was done outside in the DBI.xs. I'm using the temporary nature of $sth created inside DBD::Pg::db::do(), that is cleaning up link between $sth and $dbh needed to display the ParamValues. 

Hope that makes sense
/dk

### CODE REVIEW

File: dbdimp.c [hunk 1]
```diff
@@ -685,6 +685,15 @@ void dbd_db_destroy (SV * dbh, imp_dbh_t * imp_dbh)
...
 
 	if (TSTART_slow) TRC(DBILOGFP, "%sBegin dbd_db_destroy\n", THEADER_slow);
+ 
+	if ( imp_dbh->do_tmp_sth != NULL) {
+		imp_sth_t * imp_sth = imp_dbh->do_tmp_sth;
+		while ( imp_sth != NULL && imp_sth->do_tmp_owner == imp_dbh ) {
+			imp_sth->do_tmp_owner = NULL;
+			imp_sth = imp_sth->do_tmp_old_val;
+		}
+		imp_dbh->do_tmp_sth == NULL;
```
Comment (turnstep) [inline dbdimp.c]: s/==/=/

File: Pg.xs [hunk 2]
```diff
@@ -356,6 +356,9 @@ void do(dbh, statement_sv, attr=Nullsv, ...)
 			if (!dbdxst_bind_params(sth, imp_sth, items-2, ax+2))
 				XSRETURN_UNDEF;
 			imp_sth->async_flag = asyncflag;
+			imp_sth->do_tmp_owner = imp_dbh;
+			imp_sth->do_tmp_old_val = imp_dbh->do_tmp_sth;
```
Comment (turnstep) [inline Pg.xs]: Do we really need to store the previous sth? Each dbh should create exactly one temporary sth at a time.

File: dbdimp.c [hunk 3]
```diff
@@ -778,12 +787,14 @@ SV * dbd_db_FETCH_attrib (SV * dbh, imp_dbh_t * imp_dbh, SV * keysv)
...
 
-	case 11: /* pg_INV_READ  pg_protocol */
+	case 11: /* pg_INV_READ  pg_protocol ParamValues */
 
 		if (strEQ("pg_INV_READ", key))
 			retsv = newSViv((IV)INV_READ);
 		else if (strEQ("pg_protocol", key))
 			retsv = newSViv((IV)imp_dbh->pg_protocol);
+		else if (strEQ("ParamValues", key) && imp_dbh->do_tmp_sth != NULL)
+			return dbd_st_FETCH_attrib (dbh, imp_dbh->do_tmp_sth, keysv);
```
Comment (turnstep) [inline dbdimp.c]: Heh, nice trick

File: dbdimp.c [hunk 4]
```diff
@@ -3935,6 +3947,11 @@ void dbd_st_destroy (SV * sth, imp_sth_t * imp_sth)
 	if (NULL == imp_sth->seg) /* Already been destroyed! */
 		croak("dbd_st_destroy called twice!");
 
+	if ( imp_sth->do_tmp_owner != NULL && imp_sth->do_tmp_owner->do_tmp_sth == imp_sth) {
```
Comment (turnstep) [inline dbdimp.c]: We can probably take this out completely if we don't worry about "other" sths.

File: dbdimp.c (hunk 1 above)
Comment (dk) [reply dbdimp.c]: yes

File: Pg.xs (hunk 2 above)
Comment (dk) [reply Pg.xs]: Most probably no, but I'm being paranoid about areas I don't know. Namely, could be an execute() call inside execute()? If no, then let's remove it

**Comment by dk** [comment]
Other than this looks fine. How do we proceed from here? Could you apply the revised code, or should I file a new or updated pull request?

**Comment by turnstep** [comment]
Let's keep going the git(hub) way: make some changes to this branch and then push it out. Three things I think need addressing:
* Simplify the code and don't worry about saving/restoring previous sth. It's very short-lived and there will never be more than one.
* Pg.pm docs
* That one typo :)

**Comment by dk** [comment]
Kindly check out if it's okay now

**Comment by turnstep** [comment]
Coming along nicely! Still needs a ParamValues inside the Database Handle Attribute section.

**Comment by turnstep** [comment]
Also give yourself credit in the Changes file!

**Comment by dk** [comment]
Here's some more, please take a look

**Comment by greg-binary** [comment]
If you can clear up the conflict with the "Changes" file, I can merge this in.

**Comment by dk** [comment]
I did merge the changes for 3.8.1. Github still though shows "conflict".

**Pull request merged**